client/src/pages/ProfileSettings.js

```diff
@@ -0,0 +1,231 @@
+import React, { useEffect, useState } from "react";
+import {
+  Card,
+  Container,
+  Avatar,
+  makeStyles,
+  Typography,
+  Divider,
+  Box,
+  Grid,
+  List,
+  ListItem,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Button,
+  Input,
+} from "@material-ui/core";
+import EditOutlinedIcon from "@material-ui/icons/EditOutlined";
+import axios from "axios";
+import { useUser } from "../contexts/UserContext";
+import EditNameDialog from "../components/EditNameDialog";
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    paddingTop: theme.spacing(5),
+  },
+  card: {
+    width: "70%",
+    height: "70vh",
+    margin: "0 auto",
+    padding: theme.spacing(4),
+  },
+  heading: {
+    textAlign: "center",
+    fontSize: "2rem",
+  },
+  divider: {
+    border: `5px solid ${theme.palette.primary.main}`,
+    marginTop: theme.spacing(1),
+    width: "30%",
+    margin: "0 auto",
+  },
+  avatar: {
+    width: "100%",
+    height: "100%",
+  },
+  editIcon: {
+    // margin: "0 .5rem",
+  },
+  fileInput: {
+    display: "none",
+  },
+  userDetails: {
+    paddingLeft: theme.spacing(2),
+  },
+}));
+
+const ProfileSettings = () => {
+  const [user, setUser] = useUser();
+  const [openDialog, setOpenDialog] = useState(false);
+  const [openEditNameDialog, setOpenEditNameDialog] = useState(false);
+  const [profileImageFile, setProfileImageFile] = useState(null);
+  const [profileImageUrlInput, setProfileImageUrlInput] = useState("");
+  const [profileImageUrl, setProfileImageUrl] = useState("");
+
+  const handleClose = async () => {
+    setOpenDialog(false);
+  };
+
+  const closeEditNameDialog = () => {
+    setOpenEditNameDialog(false);
+  }
+
+  const handleFileInput = async (e) => {
+    setProfileImageFile(e.target.files[0]);
+  };
+
+  // Image is uploaded to AWS S3 using form Data
+  const handleFileUpload = async () => {
+    if (profileImageFile) {
+      const data = new FormData();
+      data.append("profileImage", profileImageFile, profileImageFile.name);
+
+      try {
+        const result = await axios.post(
+          `http://localhost:3001/edit-profile/image/${user.id}`,
```

Comment:
think you don't need to include http://localhost:3001 since its on a proxy and other areas suggest it works without it :)

Reply:
I removed localhost:3001 . thanks!